packages/client/src/client4.ts

```diff
@@ -1928,11 +1928,30 @@ export default class Client4 {
         );
     };
 
-    getPostThread = (postId: string, fetchThreads = true, collapsedThreads = false, collapsedThreadsExtended = false) => {
-        return this.doFetch<PostList>(
-            `${this.getPostRoute(postId)}/thread${buildQueryString({skipFetchThreads: !fetchThreads, collapsedThreads, collapsedThreadsExtended})}`,
+    getPostThread = async (postId: string, fetchThreads = true, collapsedThreads = false, collapsedThreadsExtended = false, direction: 'up'|'down' = 'down', perPage: number = PER_PAGE_DEFAULT, fromCreateAt?: number, fromPost?: string, prevList?: PostList): Promise<PostList> => {
+        const list: PostList = prevList || {
+            order: [postId],
+            posts: {},
+            prev_post_id: '',
+            next_post_id: '',
+        };
+
+        const result = await this.doFetch<PaginatedPostList>(
+            `${this.getPostRoute(postId)}/thread${buildQueryString({skipFetchThreads: !fetchThreads, collapsedThreads, collapsedThreadsExtended, direction, perPage, fromPost, fromCreateAt})}`,
             {method: 'get'},
         );
+
+        list.order!.push(...result.order.slice(1));
```

Comment:
Are you sure we need the `!` after `list.order`? I would've thought it's guaranteed to be defined by the type definition

Reply:
for some reason my `tsc` was yelling at me when I removed that. I will try and check if it still does!